File: 30 Days of Code/Day 10 Binary Numbers/Solution.py
# https://www.hackerrank.com/challenges/30-binary-numbers/problem

#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input())
    binaryNumber = []
    count, temp = 0, 0
    while n > 0:
        remainder = int(n % 2)
        binaryNumber.append(remainder)
        if remainder == 1:
            temp += 1
            if temp > count:
                count = temp
        else:
            temp = 0
        n = int(n / 2)
    
    print(count)

File: 30 Days of Code/Day 10 Binary Numbers/Solution.cpp
// https://www.hackerrank.com/challenges/30-binary-numbers/problem

#include <bits/stdc++.h>

using namespace std;



int main()
{
    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');
    int count = 0;
    int temp = 0;
    vector<int> binaryNumber;
    while(n>0){
        int remainder = n % 2;
        binaryNumber.push_back(remainder);
        if(remainder == 1){
            ++temp;
            if(temp > count){
                count = temp;
            }
        }
        else{
            temp = 0;
        }
        n /= 2;
    }
    cout<<count<<endl;
    return 0;
}
